liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -299,44 +288,169 @@ public ServerTask getServerTask() throws Exception {
             return artifactPaths;
         }
 
+        private Properties getPropertiesWithKeyPrefix(Properties p, String prefix) {
+            Properties result = new Properties();
+            if (p != null) {
+                Enumeration<?> e = p.propertyNames();
+                while (e.hasMoreElements()) {
+                    String key = (String) e.nextElement();
+                    if (key.startsWith(prefix)) {
+                        result.put(key, p.get(key));
+                    }
+                }
+            }
+            return result;
+        }
+
+        private List<Dependency> getEsaDependency(List<Dependency> dependencies) {
+            List<Dependency> deps = new ArrayList<Dependency>();
+            if (dependencies != null) {
+                for (Dependency d : dependencies) {
+                    if ("esa".equals(d.getType())) {
+                        deps.add(d);
+                    }
+                }
+            }
+            return deps;
+        }
+        private List<Dependency> getCompileDependency(List<Dependency> dependencies) {
+            List<Dependency> deps = new ArrayList<Dependency>();
+            if (dependencies != null) {
+                for (Dependency d : dependencies) {
+                    if ("compile".equals(d.getScope())) {
+                        deps.add(d);
+                    }
+                }
+            }
+            return deps;
+        }
+
+        private boolean hasServerPropertyChanged(MavenProject project, MavenProject backupProject) {
+            Properties p = getPropertiesWithKeyPrefix(project.getProperties(), "liberty.bootstrap.");
+            Properties q = getPropertiesWithKeyPrefix(backupProject.getProperties(), "liberty.bootstrap.");
+            if (!Objects.equals(p, q)) {
+                return true;
+            } else {
+                p = getPropertiesWithKeyPrefix(project.getProperties(), "liberty.jvm.");
+                q = getPropertiesWithKeyPrefix(backupProject.getProperties(), "liberty.jvm.");
+                if (!Objects.equals(p, q)) {
+                    return true;
+                } else {
+                    p = getPropertiesWithKeyPrefix(project.getProperties(), "liberty.env.");
+                    q = getPropertiesWithKeyPrefix(backupProject.getProperties(), "liberty.env.");
```

Comment:
No reason to call project.getProperties() three times and backupProject.getProperties() three times. Get them before their first use and store in variable. Also no need for else statements since the if statements return true.

Reply:
ok